sap/cli/checkout.py

```diff
@@ -43,6 +45,25 @@ def dump_attributes_to_file(object_name, abap_attributes, typsfx, ag_serializer,
         writer.close()
 
 
+def write_source(source, object_name, typsfx, destdir=None):
+    """Write source code to file"""
+
+    filename = build_filename(object_name, typsfx, 'abap', destdir=destdir)
+    with open(filename, 'w', encoding='utf8') as dest:
+        dest.write(source)
+
+
+def download_abapgit_source(object_name, source_object, typsfx, destdir=None):
+    """Saves the source code of object in AbapGit format"""
+
+    try:
+        source_code = source_object.get_abapgit_source()
```

Comment:
I would rather have checkout configurable - the source formatter should be an object - either abapGit or in future  ABAP File Format https://github.com/SAP/abap-file-formats